src/map/lua/lua_baseentity.cpp

```diff
@@ -16660,6 +16679,40 @@ bool CLuaBaseEntity::hasTPMoves()
     return !MobSkills.empty();
 }
 
+/************************************************************************
+ *  Function: drawIn()
+ *  Purpose : Draws in the target, or current target if not specified
+ *  Example : mob:drawIn()     mob:drawIn(player)
+ *  Notes   : Draws in a player even if within the draw-in leash
+ ************************************************************************/
+void CLuaBaseEntity::drawIn(sol::variadic_args va)
+{
+    auto mobObj = dynamic_cast<CMobEntity*>(m_PBaseEntity);
```

Comment:
This could be called accidentally on a player or an NPC, then mobObj would be `nullptr`, and then there's no check or logging to say what's going on. Then if va.size() == 0 it'll get dereferenced and there'll be a crash

Reply:
This check at the start of the function should correct the wrong entity type calling this, and/or the de-referencing of the nullptr

```cpp
    if (m_PBaseEntity->objtype != TYPE_MOB)
    {
        ShowError("Attempting to draw-in from non-mob entity: %s", m_PBaseEntity->getName());
        return;
    }
```